packages/components/src/box-control/all-input-control.tsx

```diff
@@ -69,18 +83,43 @@ export default function AllInputControl( {
 		} );
 	};
 
+	const [ parsedQuantity, parsedUnit ] =
+		parseQuantityAndUnitFromRawValue( allValue );
+
 	return (
-		<UnitControl
-			{ ...props }
-			disableUnits={ isMixed }
-			isOnly
-			value={ allValue }
-			onChange={ handleOnChange }
-			onUnitChange={ handleOnUnitChange }
-			onFocus={ handleOnFocus }
-			onHoverOn={ handleOnHoverOn }
-			onHoverOff={ handleOnHoverOff }
-			placeholder={ allPlaceholder }
-		/>
+		<HStack>
+			<UnitControl
+				{ ...props }
+				disableUnits={ isMixed }
+				id={ inputId }
+				isOnly
+				value={ parsedQuantity }
```

Comment:
I was trying to keep it closer to trunk: 

https://github.com/WordPress/gutenberg/blob/59e89580e4a0e5b9fbf68d07470a339bc8fd7949/packages/components/src/box-control/all-input-control.tsx#L77

I thought it may be implemented this way because the units can be disabled when `isMixed` is true. But then we could check against that if it was the case.  😅 So to be honest, I'm not 100% sure, I just was trying to make as few changes as possible, but I needed the unit for RangeControl. 

I can look into it if it seems like a good change to include here.